micro-rdk/src/esp32/camera.rs

```diff
@@ -51,32 +85,28 @@ impl Esp32Camera {
             last_grab: t.now(),
         }
     }
-    pub fn setup(&self) -> Result<(), CameraError> {
-        let ret = (unsafe { crate::esp32::esp_idf_svc::sys::esp_camera_init(&self.config) })
-            as crate::esp32::esp_idf_svc::sys::esp_err_t;
+    fn setup(&self) -> Result<(), CameraError> {
+        let ret =
+            (unsafe { esp_camera_init(&self.config) }) as crate::esp32::esp_idf_svc::sys::esp_err_t;
         let ret = crate::esp32::esp_idf_svc::sys::EspError::convert(ret);
-        ret.map_err(|e| CameraError::CameraInitError(e.into()))
+        ret.map_err(|e| CameraError::InitError(e.into()))
     }
-    pub fn get_cam_frame(&self) -> Option<*mut crate::esp32::esp_idf_svc::sys::camera_fb_t> {
-        let ptr = (unsafe { crate::esp32::esp_idf_svc::sys::esp_camera_fb_get() })
-            as *mut crate::esp32::esp_idf_svc::sys::camera_fb_t;
+    pub fn get_cam_frame(&self) -> Option<*mut camera_fb_t> {
+        let ptr = (unsafe { esp_camera_fb_get() }) as *mut camera_fb_t;
         if ptr.is_null() {
             None
         } else {
             Some(ptr)
         }
     }
-    pub fn return_cam_frame(
-        &self,
-        frame: Option<*mut crate::esp32::esp_idf_svc::sys::camera_fb_t>,
-    ) {
+    pub fn return_cam_frame(&self, frame: Option<*mut camera_fb_t>) {
```

Comment:
Is there an obligation to call `return_cam_frame` for each call to `get_cam_frame`? Could you build a resource management type to ensure that automatically?